### PR TITLE
feat: Add Agent Skill for NEL config generation, running evals and progress monitoring

### PR DESCRIPTION
**Pull request by gchlebus**
### Summary

Adds a comprehensive Claude skill (`nel-assistant`) for generating NeMo Evaluator Launcher (NEL) configuration files through an interactive workflow, along with a `nel skills` CLI subcommand for installing skills and building configs from templates.

### Added

- **`nel-assistant` skill** (`packages/nemo-evaluator-launcher/.claude/skills/nel-assistant/`)
  - `SKILL.md` (282 lines) — detailed skill instructions covering configuration generation, running evaluations, monitoring progress, multi-node setup, known issues, and documentation errata
  - Config-templates reference plan (`.claude/plans/config-templates-reference.md`) for the `make update-configs` consistency checker

- **`nel skills` CLI subcommand** (`cli/skills.py`, 254 lines)
  - `nel skills install` — installs NEL agent skills into Claude Code, Cursor, Codex, or OpenCode (supports `--project` for per-project install)
  - `nel skills build-config` — builds a complete evaluation config by merging YAML template excerpts

- **`build_config` module** (`common/build_config.py`, 143 lines)
  - Merges YAML asset files (deployment + evaluation + execution + export) into a single config
  - Smart `resolve_output_path` that fills gaps in numbered output directories

- **YAML config templates** (`resources/config_templates/`)
  - Deployment backends: NIM, vLLM, SGLang, TRT-LLM, none
  - Evaluation types per mode (base/chat/reasoning): code, default, math_reasoning, multilingual, safety, standard
  - Execution modes: local, Slurm
  - Export destinations: MLflow, W&B, none

- **Test suite** (`tests/unit_tests/test_build_config.py`, 592 lines)
  - Covers `build_config`, `resolve_output_path` (including gap-filling in numbered filenames), and template merging edge cases

- **New example** — `slurm_vllm_multinode_dp_haproxy.yaml`: multi-node data parallel with HAProxy load balancing

- **New eval test case** — `nemotron3-nano-bf16-reasoning.json` for Nemotron 3 Nano BF16

- **Claude plugin manifests** (`.claude-plugin/`)
  - `plugin.json` and `marketplace.json` for skill discoverability via Claude Code marketplace

- **`make update-configs`** — config consistency checker that diffs templates against `main` and pipes to Claude for review

### Changed

- **README.md** — new "Agentic Skills" section with install instructions (`nel skills install`, marketplace)
- **`docs/evaluation/run-evals/reasoning.md`** — added sglang documentation link, added section on enabling vLLM reasoning parser via `--reasoning-parser` / `deployment.extra_args`, removed duplicated "Additional resources" section
- **Examples** — minor annotation improvements: interceptor ordering comment in `local_advanced_adapters.yaml`, `log_config_params: true` in `local_auto_export.yaml`, minor tweaks in Slurm multinode examples
- **`makefile`** — added `update-configs` phony target

### CODE REVIEW

File: packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/resources/config_templates/deployment/none.yaml
Comment (gchlebus) [inline packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/resources/config_templates/deployment/none.yaml:1]: How do we plan to keep skill assets up to date? Why not provide pointers to relevant docs/source code locations so that the model can pick up the most recent info on its own?

File: packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/resources/config_templates/export/mlflow.yaml
Comment (gchlebus) [inline packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/resources/config_templates/export/mlflow.yaml:1]: I think we should include also csv exporter if it's not the case already.

File: packages/nemo-evaluator-launcher/.claude/skills/nel-assistant/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+   Combine with `-o` for limited samples: `nel run --config <config_path> -t <task_name> -o ++evaluation.nemo_evaluator_config.config.params.limit_samples=10`
+
+4. **Full evaluation** (production run):
+   ```
+   nel run --config <config_path>
+   ```
+
+After the dry-run, check the output from `nel` for any problems with the config. If there are no problems, propose to first execute the test run with limited samples and then execute the full evaluation. If there are problems, resolve them before executing the full evaluation.
+
+**Monitoring Progress**
```
Comment (gchlebus) [inline packages/nemo-evaluator-launcher/.claude/skills/nel-assistant/SKILL.md:213]: This seems to be outside of the skill scope indicated by its name. If you think this skill could be extended to cover also monitoring of jobs, we should probably adjust its name.

File: packages/nemo-evaluator-launcher/.claude/skills/nel-assistant/scripts/verify_config.py
Comment (gchlebus) [inline packages/nemo-evaluator-launcher/.claude/skills/nel-assistant/scripts/verify_config.py:1]: How skill specific is this validation logic? I think that this functionality could be moved to the "main" package and we could use it to validate configs and provide informative error messages to users when executing `nel` commands requiring configs.

Comment (piojanu) [reply packages/nemo-evaluator-launcher/.claude/skills/nel-assistant/scripts/verify_config.py:1]: Yes, I'd love to do it. But this is a discussion for our exec meeting.

File: packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/resources/config_templates/deployment/none.yaml
Comment (piojanu) [reply packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/resources/config_templates/deployment/none.yaml:1]: We [set appropriate degree of freedom](https://platform.claude.com/docs/en/agents-and-tools/agent-skills/best-practices#set-appropriate-degrees-of-freedom) here for the agent to effectively approach a fragile and error-prone task. It allows the agent to be much more reliable and start quickly (without the need for extensive exploration of our docs or codebase each new session).

This is how a human engineer would do it ;) Start with some example and change it to their needs. But in here we introduce a solution to build an example so we cover all possible base cases.

When it comes to the maintenance, I plan to move it to the main codebase as the NEL CLI. And assets we need to maintain in the same way as we maintain other example configs. I want to introduce a `make update-configs` that will trigger an agent with the appropriate plan that I've already used to create these assets.

File: packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/resources/config_templates/export/mlflow.yaml
Comment (piojanu) [reply packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/resources/config_templates/export/mlflow.yaml:1]: You mean `local` exporter, right? I don't think it makes much sense with auto-export. It'd just move things around e.g. on the cluster.

So I'd leave it as is and if someone want's to export to local, they use the `nel export` config.

File: packages/nemo-evaluator-launcher/.claude/skills/nel-assistant/SKILL.md (hunk 1 above)
Comment (piojanu) [reply packages/nemo-evaluator-launcher/.claude/skills/nel-assistant/SKILL.md:213]: Renamed

**Comment by gchlebus** [comment]
LGTM 👍
